docs/meeting-notes/foam-core-2020-07-11.md

```diff
@@ -0,0 +1,125 @@
+# Foam Core 2020-07-11
+
+Present: @jevakallio, @riccardoferretti
+
+### Tests
+
+- How do we know this approach works?
+  - Supports renaming
+  - Supports searching with (attribute-x)
+  - Find dead links
+
+### Getting started
+
+- Land work to master
+  - Create a foam-core package
+  -
+
+### Open questions
+
+- How should writing to files work
+  - What if affected notes have unsaved changes
+
+### Graph methods
+
+- get all
+- search by
+  - tag
+  - free text
+  - [[todo]]: how do vs code search editors work? are they pluggable? what do they need?
+- find dead links
+  - for linters
+- serialize/toJSON (for visualizers)
+- subscribe to changes
+
+### Node methods
+
+- rename
+- get links
+  - forward links (for link lists)
+  - backlinks (with surrounding context)
+
+### Node definition
+
+What do we need the node (and edge metadata) to contain:
+
+- `id`: tbd
+  - should be unique, needs some kind of unique gen function
+  - should be reconstructable even if links are not updated every time
+  - what happens during rename? is reparenting the graph going to be hard?
```

Comment:
What does the reparenting of the graph mean in this context?

Reply:
I believe my thinking here was that if we use file path as id, and the file's path changes, is it possible to update the node's id, or must the node be deleted, new Note node be created, and then all in/outEdges be re-routed to it. 

If id is mutable, then this is no issue.

If it's not, it's merely an inconvenience.

Reply:
I was leaning towards having the ID being the URI of the file, basically case n.2
I don't see it as a big issue, and I think it might be cleaner, but I am open to the idea of having another ID for the note (the question would be how to generate it/store it/....)